examples/secp256k1-transfer/src/lib.ts

```diff
@@ -0,0 +1,108 @@
+  const pubKey = hd.key.privateToPublic(privKey);
+  const args = hd.key.publicKeyToBlake160(pubKey);
+  const template = AGGRON4.SCRIPTS["SECP256K1_BLAKE160"]!;
+  const lockScript: Script = {
+    codeHash: template.CODE_HASH,
+    hashType: template.HASH_TYPE,
+    args: args,
+  };
+  const address = helpers.encodeToAddress(lockScript, { config: AGGRON4 });
+  return {
+    lockScript,
+    address,
+    pubKey,
+  };
+};
+
+/**
+ * calculate the transaction fee by skeleton inputs and outputs
+ * @param skeleton unsigned transaction skeleton
+ * @returns the transaction fee, it is from `sum(skeleton inputs capacity) - sum(skeleton outputs capacity)`
+ */
+export function getPaidTransactionFee(skeleton: helpers.TransactionSkeletonType) {
+  const inputs = skeleton.inputs.reduce((acc, cur) => acc.add(cur.cellOutput.capacity), BI.from(0));
+  const outputs = skeleton.outputs.reduce((acc, cur) => acc.add(cur.cellOutput.capacity), BI.from(0));
+  return inputs.sub(outputs);
+}
+/**
+ * fetch all cells and calculate the sum of their capacities
+ */
+export async function fetchAddressBalance(address: string): Promise<BI> {
+  let balance = BI.from(0);
+
+  for await (const cell of indexer.collector({ lock: helpers.parseAddress(address, { config: AGGRON4 }) }).collect()) {
+    balance = balance.add(cell.cellOutput.capacity);
+  }
+
+  return balance;
+}
+
+interface Options {
+  targets: {
+    address: string;
+    capacity: BIish;
+  }[];
+  address: string;
+}
+
+/**
+ * create an unsigned transaction skeleton which includes several inputs and outputs(for multiple transaction receivers)
+ */
+export async function createUnsignedTxSkeleton(options: Options) {
+  let txSkeleton = helpers.TransactionSkeleton({ cellProvider: indexer });
+  for (const target of options.targets) {
+    txSkeleton = await commons.common.transfer(
+      txSkeleton,
+      [options.address],
+      target.address,
+      target.capacity,
+      options.address,
+      undefined,
+      { config: AGGRON4 }
+    );
+  }
+
+  txSkeleton = await payFeeByFeeRate(txSkeleton, [options.address], 1000, undefined, { config: AGGRON4 });
+  return txSkeleton;
+}
+
+/**
+ * sign a transaction skeleton
+ * @param txSkeleton unsigned transaction skeleton
+ * @param privateKey the private key which can unlock input cells
+ * @returns
+ */
+export function signTransaction(txSkeleton: helpers.TransactionSkeletonType, privateKey: string) {
+  txSkeleton = commons.common.prepareSigningEntries(txSkeleton);
+  const message = txSkeleton.get("signingEntries").get(0)?.message;
+  const signature = hd.key.signRecoverable(message!, privateKey);
+  const tx = helpers.sealTransaction(txSkeleton, [signature]);
+  return tx;
+}
+
+/**
+ * send a transaction to CKB testnet
+ * @returns Promise with transaction hash
+ */
+export async function transfer(tx: Transaction): Promise<string> {
+  const hash = await rpc.sendTransaction(tx, "passthrough");
```

Comment:
does this `rpc` need to be passed in from the function parameter?

Reply:
I think it's no need. because the definition in https://github.com/nervosnetwork/lumos/pull/388/files#diff-23baadf37b78885a26cae62ec8dff08aa7dbd4083b9df75e84593d6adb404fa0R9 tells me it's a singleton.